### DISCUSSION TITLE
UHO openings contain many too imbalanced positions

### DISCUSSION THREAD

**Discussion by PavelChess** [official-stockfish/Stockfish#5079] [General]
The purpose of imbalanced opening was to reduce draws percentage. The expected outcome was that successful patch will increase wins/losses ratio.

I have ran handicapped match of 2 copies of recent version. One copy got search limit of 200000 nodes, and other of 100000. Openings book UHO_Lichess_4852_v1.epd.
The match ended by result 416 wins - 122 losses - 462 draws.

So about 25% of positions from first 500 of the book are so imbalanced that even handicap 2:1 couldn't give pair result 1.5-0.5.

Usually good patch can give +0.5 to +2 ELO. This is much less than enough to avoid loss at those positions that 2:1 handicap had no success.

This can lead to wrong or unwanted testing results, for example:
1. Gaining patches test needs much more games
2. Gaining ELO patch can fail because of too low gain
3. Simplification patch can pass because a test shows less ELO loss than real.
4. Simplification patch that fails needs much more games
5. Influence of this at LTC may be greater than at STC

I suggest to clean up imbalanced opening positions that even handicap for example 2:1 cannot avoid losses at them

**Comment by robertnurnberg** [comment]
Thanks for sharing your data. Just to say that we do have this on our radar and are tracking the sharpness of the book exits on the Chess Cloud Database [chessdb.cn](https://chessdb.cn/queryc_en/), see the repo [UHOtrack](https://github.com/robertnurnberg/uhotrack). So in future we may well filter some of the positions from the current book. For now we do not see an urgent need to change the book, though.

**Comment by vdbergh** [comment]
@PavelChess If you suspect the book can be improved then you should make a new version and test it according to the procedure described here https://github.com/official-stockfish/fishtest/wiki/Fishtest-faq#how-to-compare-opening-books
